### PR TITLE
Feature/aql-keep-recursive

### PR DESCRIPTION
**Pull request by Simran-B**
- Add documentation for the new AQL document function `KEEP_RECURSIVE()`
  https://github.com/arangodb/arangodb/pull/15923
- Convert examples in `functions-document.md` to @EXAMPLE_AQL but only in 3.10
- Improve some of the other function descriptions and examples, as well as the formatting following
  #878

https://jenkins.arangodb.biz:3456/view/Documentation/job/arangodb-ANY-examples/1781/

Preview: https://deploy-preview-958--zealous-morse-14392b.netlify.app/docs/3.10/aql/functions-document.html

### CODE REVIEW

**Review by ansoboleva — APPROVED** [review]
reviewed

File: 3.10/aql/functions-document.md [hunk 1]
```diff
@@ -345,142 +622,365 @@ individual parts as separate attributes.
...
-```
+**Examples**
+
+Recursively remove `baz` attributes, by passing an array with the attribute key:
+
+    {% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline aqlUnsetRecursive_7
+    @EXAMPLE_AQL{aqlUnsetRecursive_7}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, ["baz"])
```
Comment (KVS85) [inline 3.10/aql/functions-document.md:920]: Looks like a meaningless example comparing to original `UNSET_RECURSIVE( doc, [ "_id", "_key", "foo", "bar" ] )`.

**Review by KVS85 — COMMENTED** [review]
LGTM.

File: 3.10/aql/functions-document.md (hunk 1 above)
Comment (Simran-B) [reply 3.10/aql/functions-document.md:920]: The KEEP_RECURSIVE and UNSET_RECURSIVE examples use the same example object and I don't really want to use an object that has nested, system attribute-like keys, which would be the only way to make it meaningful for the different query variations, while still using the original arguments for the function call here.

The point of the specific example is to show that you can pass an array. For the keep counterpart it made sense to show passing multiple keys but here it seemed better to let some attributes remain in the object.

I agree that many of the examples are contrived but making them more real-world would also make them less succinct.

**Pull request merged**